sdk/eventhub/Azure.Messaging.EventHubs/samples/Program.cs

```diff
@@ -148,19 +184,37 @@ private static ISample RetrieveSample()
             {
                 var identitySamples = LocateSamples<IEventHubsIdentitySample>();
 
-                PrintEventHubsIdentitySamples(identitySamples);
-
                 while (IsEventHubsIdentity(samples, choice))
                 {
+                    PrintEventHubsIdentitySamples(identitySamples);
                     choice = ReadSelection(identitySamples);
 
                     if (choice.HasValue && !IsGoBack(identitySamples, choice))
                     {
                         return identitySamples[choice.Value];
                     }
-                    else if (IsGoBack(identitySamples, choice))
+                    if (IsGoBack(identitySamples, choice))
                     {
-                        choice = ReadSelection(samples);
+                        return RetrieveSample();
```

Comment:
We need to 'start over' when we select Go Back. Otherwise, pressing 12 - 2 - 13 would break because you're stuck in this loop of the same kind of samples, and 13 (SR) samples are not 12 (Identity).

Reply:
Agreed.   The former method was not designed to be robust.   In the (hopefully) not so distant future, I'm going to break apart samples into their own applications and kill off this bit.